coda/coda_mdstore/views.py

```diff
@@ -532,77 +532,76 @@ def bagHTML(request, identifier):
     """
 
     # assign the proxy url
     proxyRoot = "http://%s/" % request.META.get('HTTP_HOST')
-    pathList = []
-    transList = []
     # attempt to grab a bag,
     try:
         Bag.objects.get(name=identifier)
     except Bag.DoesNotExist:
         return HttpResponseNotFound(
             "There is no bag with id {0}.".format(identifier)
         )
-    handle = getFileHandle(identifier, "manifest-md5.txt")
-    if not handle:
+    transList = generateBagFiles(identifier, proxyRoot, settings.CODA_PROXY_MODE)
+    if transList == 'Http404':
         raise Http404
-    bag_root = handle.url.rsplit('/', 1)[0]
-    line = handle.readline()
-    # iterate over handle and append urls to pathlist
-    while line:
-        line = line.strip()
-        parts = line.split(None, 1)
-        if len(parts) > 1:
-            pathList.append(parts[1])
-        line = handle.readline()
-    # iterate top files and append to pathlist
-    try:
-        topFileHandle = getFileHandle(identifier, "")
-        topFiles = getFileList(topFileHandle.url)
-        for topFile in topFiles:
-            pathList.append(topFile)
-    except:
-        pass
-    # iterate pathlist and resolve a unicode path dependent on proxy mode
-    for path in pathList:
-        if isinstance(path, bytes):
-            try:
-                path = path.decode()
-            except UnicodeDecodeError:
-                path = path.decode('latin-1')
-        # CODA_PROXY_MODE is a settings variable
-        if settings.CODA_PROXY_MODE:
-            uni = '%sbag/%s/%s' % (
-                proxyRoot, identifier, path
-            )
-        else:
-            uni = bag_root + "/" + path
-        # throw the final path into a list
-        transList.append(uni)
-
-    if html:
-        return render(request, 'mdstore/bag_files_download.html',
-                      {'links': sorted(transList)})
-
-    if download:
-        meta_id = identifier.split('/')[-1]
-        zip_filename = meta_id + '.zip'
-        response = StreamingHttpResponse(zip_file_streamer(transList, meta_id),
-                                         content_type='application/zip')
-        response['Content-Disposition'] = 'attachment; filename=%s' % zip_filename
-        return response
 
     outputText = "\n".join(reversed(transList))
     resp = HttpResponse(outputText, content_type="text/plain")
     resp.status_code = 200
     return resp
 
 
+def bagURLLinks(request, identifier):
+    """
+        Return links to URLS in the bag
+    """
+    # assign the proxy url
+    proxyRoot = "http://%s/" % request.META.get('HTTP_HOST')
+    # attempt to grab a bag,
+    try:
+        Bag.objects.get(name=identifier)
+    except Bag.DoesNotExist:
+        return HttpResponseNotFound(
+            "There is no bag with id {0}.".format(identifier)
+        )
+    transList = generateBagFiles(identifier, proxyRoot, settings.CODA_PROXY_MODE)
+    if transList == 'Http404':
+        raise Http404
```

Comment:
Since `generateBagFiles` is meant to return a list of files and not in itself trigger a 404 page response, I might lean toward raising the `Http404` in the view still and, instead of returning "Http404" as a string, perhaps raise some other exception that is caught in the view.

Regarding the second point, according to the [docs](https://docs.djangoproject.com/en/2.2/topics/http/views/#the-http404-exception):

>  If you raise Http404 at any point in a view function, Django will catch it and return the standard error page for your application, along with an HTTP error code 404.

Does that mean we wouldn't be able to catch the exception ourselves in the view logic?

Reply:
I don't _think_ that's what it means, I think there's an unwritten line of "if it's not caught elsewhere first", but I might be wrong about that.

Reply:
I thought raising Http404 error from the django view would be more appropriate and hence returned str 'Http404' from `generateBagFiles`. I can raise an exception here and catch that in the view as @ldko suggested.

Reply:
I think this is looking better, I'm just wondering if we should be creating our own exception to be raised, or maybe use something more specific than `Exception`, but I'm also ok with leaving it as-is

Reply:
After looking again, we use both custom exceptions and plain Exceptions, so really I'm fine either way. My preference is towards more focused, custom exceptions, but I think it kind of becomes a rabbit-hole of "if we make this one, maybe we should also use it here, or make another custom exception there"

Reply:
Looking at this now, I do think it would be good to make a specific exception to catch in these views that call `generateBagFiles` when raising a 404 because it is a relatively long function, and if, down the line, something else in it unexpectedly throws some other random exception it would bubble up here as a 404 when it may be something that needs fixing.